mods/flowers/init.lua

```diff
@@ -644,15 +644,21 @@ if not flowers.reg3 then
 
 		on_place = function(itemstack, placer, pointed_thing)
 			local pos = pointed_thing.above
-			local node = minetest.get_node(pointed_thing.under).name
-			local def = minetest.reg_ns_nodes[node]
+			local node = minetest.get_node(pointed_thing.under)
+			local name = node.name
+			local def = minetest.reg_ns_nodes[name]
 			local player_name = placer:get_player_name()
 
+			-- Pass through interactions to nodes that define them (like chests).
+			if def.on_rightclick and not placer:get_player_control().sneak then
+				return def.on_rightclick(pointed_thing.under, node, placer, itemstack, pointed_thing)
```

Comment:
```suggestion
				return def.on_rightclick(pointed_thing.under, node, placer, itemstack, pointed_thing) or itemstack
```

Reply:
This isn't actually necessary, according to API docs, returning `nil` is enough.

Reply:
righttt

Reply:
All suggestion added after re-opening the issue. :)